### ISSUE TITLE
False positive on CKV_AWS_59 - APIGatewayAuthorization ignores API keys

### ISSUE THREAD

**Issue by ketumbra** [bridgecrewio/checkov#960]
**Describe the bug**
**CKV_AWS_59** assumes the API is open to the public if authorization is **NONE**. However, API keys are another option on REST APIs and are configured separately. See https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-setup-api-key-with-console.html.

**To Reproduce**
The following terraform code will cause a test failure, which appears to be against the spirit of the rule:
```terraform
resource "aws_api_gateway_method" "POST" {
...
authorization = NONE
api_key_required = true
...
}
````

**Expected behavior**
I would expect this configuration to be considered secure. 

**Desktop (please complete the following information):**
 - OS: linux/amd64 Docker image bridgecrew/checkov:1.0.833
 - Checkov Version 1.0.833

**Comment by ketumbra** [comment]
Sure. Here goes.

**Comment by ketumbra** [comment]
aha, think that got it. Unit tests are passing.

**Comment by schosterbarak** [comment]
Thank you @ketumbra
